### DISCUSSION TITLE
[1장]퀴즈

### DISCUSSION THREAD

**Discussion by codefug** [FE-tech-talk/TechTalk-React#7] [Q&A] [answered]
// 문제 1 
```js
var hello = {
	greet: 'hello, world';
}
var hi = {
	greet: 'hello, world';
}

console.log(hello === hi); // false
console.log(hello.greet === hi.greet) // true
```
예제에서 hello와 hi가 같은 값을 가진 객체일 때 hello == hi 는 false인데 hello.프로퍼티 == hi.프로퍼티 는 true인 이유

// 문제 2
호이스팅은 함수에 대한 선언을 실행 전에 미리 메모리에 등록하는 작업을 의미한다.
실행 컨텍스트에서 호이스팅을 일으키는 요소는?

**Comment by easyhyun00** [comment]
## 답변

1. 객체는 참조형이라 다른 곳을 가리키기 때문에 false, 반면 프로퍼티는 같은 문자열이므로 true
2. var?

**Comment by codefug** [comment, marked as answer]
1. 객체 자체는 같은 값을 갖고 있어도 다른 선언이면 다른 메모리 공간을 사용하게 된다. 하지만 그 프로퍼티는 변수로써 값이 같다면 해당 값이 들어있는 메모리 공간을 공유한다.
2. 실행 컨텍스트 실행시 VariableEnvironment의 EnvironmentRecord가 미리 함수 선언문, 변수명을 메모리에 등록해놓는다.
